### PR TITLE
UI/profile

### PR DESCRIPTION
**Pull request by Keith-Go**
## Ticket
- [UI profile](https://trello.com/c/VQ5qPjKp/32-ui-profile)

## Video

https://github.com/luke-nguyen-goldenowl/join_me/assets/152830147/dde2e7c5-2d41-45dc-b10f-4ada2c27854f

### CODE REVIEW

File: lib/src/features/account/profile/logic/profile_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    final XFile? image = await picker.pickImage(source: ImageSource.gallery);
+    return image;
+  }
+
+  Future<XFile?> _pickImageToCamera(ImagePicker picker) async {
+    final XFile? image = await picker.pickImage(source: ImageSource.camera);
+    return image;
+  }
+
+  void pickImage(ImageSource source) async {
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/account/profile/logic/profile_bloc.dart]: use XImagePicker already in the source code when init project

File: lib/src/features/account/profile/view/edit_profile_view.dart [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import 'package:myapp/widgets/appbar/app_bar_custom.dart';
+import 'package:myapp/widgets/forms/input.dart';
+
+class EditProfileView extends StatelessWidget {
+  const EditProfileView({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+        create: (_) => ProfileBloc()
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/account/profile/view/edit_profile_view.dart]: init in bloc or create factory for state

File: lib/src/features/account/profile/widget/change_image_widget.dart [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            width: 100,
+            decoration: BoxDecoration(
+              borderRadius: BorderRadius.circular(15),
+              border: Border.all(
+                color: AppColors.grey,
+                width: 2,
+                style: BorderStyle.solid,
+              ),
+            ),
+            child: context.watch<ProfileBloc>().state.avatar != null
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/account/profile/widget/change_image_widget.dart]: Có thể dùng BlocBuilder cho từng chỗ để gọi state thay vì gọi context.watch<ProfileBloc>().state khá dài, mà nhớ đặt buildWhen mỗi khi thêm 1 BlocBuilder

File: lib/src/features/account/profile/widget/change_image_widget.dart [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+          ),
+          minimumSize: const Size(50, 50)),
+      onPressed: handlePress,
+      icon: icon,
+      iconSize: 30,
+    );
+  }
+}
+
+void _handleSetImage(
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/account/profile/widget/change_image_widget.dart]: Create a file with class XBottomSheet to call this function. Logic cannot be placed in the UI file. You can create a function with custom UI as the input Widget. The UI part can be created separately in another file in the widget folder.

File: lib/src/features/account/view/account_view.dart [hunk 5]
```diff
@@ -16,32 +18,44 @@ class AccountHomeView extends StatelessWidget {
     return BlocBuilder<AccountBloc, AccountState>(
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/account/view/account_view.dart:18]: add buildWhen

File: lib/src/features/account/widget/list_event_favorite.dart [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+class ListEventFavorite extends StatelessWidget {
+  const ListEventFavorite({super.key, required this.userId});
+  final String userId;
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (_) => ListEventFavoriteBloc()..setUserId(userId),
+      child: BlocBuilder<ListEventFavoriteBloc, ListEventFavoriteState>(
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/account/widget/list_event_favorite.dart:17]: add buildWhen

File: lib/src/features/account/widget/list_follower.dart [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+class ListFollower extends StatelessWidget {
+  const ListFollower({super.key, required this.userId});
+  final String userId;
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (_) => ListFollowerBloc()..setUserId(userId),
+      child: BlocBuilder<ListFollowerBloc, ListFollowerState>(
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/account/widget/list_follower.dart:19]: add buildWhen

**Comment by Keith-Go** [comment]
resolved

**Comment by luke-nguyen-goldenowl** [comment]
Approved

**Pull request merged**